src/locale/localed.c

```diff
@@ -395,6 +398,28 @@ static int method_set_locale(sd_bus_message *m, void *userdata, sd_bus_error *er
         for (LocaleVariable p = 0; p < _VARIABLE_LC_MAX; p++)
                 free_and_replace(c->locale[p], new_locale[p]);
 
+        /* Generate locale in case the system is using locale-gen */
+        if (use_localegen) {
+                r = locale_gen_enable_locale(c->locale);
+                if (r < 0) {
+                        if (r != -EINVAL) {
+                                log_error_errno(r, "Failed to enable locale generation: %m");
+                                return sd_bus_error_set_errnof(error, r, "Failed to enable locale generation: %m");
+                        }
+                } else {
+                        r = locale_gen_run();
+                        if (r != 0) {
+                                log_error_errno(r, "Failed to generate locale: %m");
+                                return sd_bus_error_set_errnof(error, r, "Failed to generate locale: %m");
+                        }
+                }
+
+                /* Sanity check to ensure the new locale is installed and usable now */
+                if (locale_is_installed(c->locale[VARIABLE_LANG]) <= 0)
```

Comment:
maybe log somethings ot system logs here too, explaining that locale-gen is fucked